test/functional/testlib.bash

```diff
@@ -1188,6 +1201,42 @@ update_manifest() { # swupd_function
 
 }
 
+get_entry_from_manifest() { # swupd_function
+
+	local manifest=$1
+	local entry_key=$2
+	local value
+
+	# If no parameters are received show usage
+	if [ $# -eq 0 ]; then
+		cat <<-EOM
+			Usage:
+			    get_entry_from_manifest <manifest> [file_hash or file_name]
+			EOM
+		return
+	fi
+	validate_item "$manifest"
+
+	# if no key is provided, return a list with all
+	# files/entries from the manifest
+	if [ -z "$entry_key" ]; then
+		sudo cat "$manifest" | grep -E -x "(D|F|L)..."$'\t'".*"$'\t'".."$'\t'"/.*" | awk '{ print $4 }'
+		return
+	fi
+
+	# a key was provided, try with the key as a file
+	# path first which should be unique in a Manifest
+	value=$(sudo cat "$manifest" | grep "$entry_key"$)
+
+	# if an entry was not found using the file path, try with the hash next
+	if [ -z "$value" ]; then
+		value=$(sudo cat "$manifest" | grep ^"...."$'\t'"$entry_key")
```

Comment:
It looks like the output format is different here than when no parameter is used. I think we need to be consistent in the format of the output

Reply:
you are right, this is because I first I had two separate functions, and when I merged them I forgot about consistency. I'll fix that too.